delivery/services/organise_service.py

```diff
@@ -207,3 +210,94 @@ def organise_sample_file(self, sample_file, organised_sample_path, lanes):
             read_no=sample_file.read_no,
             is_index=sample_file.is_index,
             checksum=sample_file.checksum)
+
+    def parse_yaml_config(config_file_path, input_key, input_value):
+        """
+        Read a configuration file for organizing a runfolder or analyzed project prior to delivery. 
+        This function will do string substitutions based on the input, keys in the configuration file
+         and results from resolved regular expressions (based on patterns from the configuration file). 
+
+        :param config_file_path: Full path to a configuration file in yaml-format.
+        :param input_key: projectid or runfolder_name
+        :param input_value: A runfolder name or a project name (for analysed projects).
+        :return: A list with one element per link/copy operation to be performed during organisation.
+         Each element consist of a tuple with 3 values ("path/to/source/file", "path/to/destination/file", "options as dictionary").
+        """
+
+        with open(config_file_path) as f:
+            config = yaml.load(f, Loader=yaml.SafeLoader)
+
+        input_dict = {}
+        input_dict[input_key] = input_value
+        input_dict.update(config["variables"]) #The input variable must be the first occuring key.
+        config["variables"] = input_dict
+        variables_dict = config["variables"]
+
+        config_as_list = []
+
+        for key in variables_dict.keys():
+            variables_dict[key] = variables_dict[key].format_map(Default(variables_dict))
+
+        for element in config["files_to_organise"]:
+            element["source"] = element["source"].format_map(Default(variables_dict))
+            element["destination"] = element["destination"].format_map(Default(variables_dict))
+
+            if "filter" in element["options"].keys(): 
+                element["options"]["filter"] = element["options"]["filter"].format_map(Default(variables_dict))
+                source_list = OrganiseService.resolve_config_regexp(element["options"]["filter"])
+                for source_file in source_list:
+                    source = ''.join([element["source"].strip('*'),source_file["source_file"]])
+                    destination_file = element["destination"].format_map(Default(source_file))
+                    destination = ''.join([destination_file, source_file["source_file"].split("/")[-1]])
```

Comment:
I don't think this is what we want in the end but we may need to look into that more closely when we look at real examples.

Also, I'd recommend:
```suggestion
                    destination = os.path.join(destination_file, os.path.basename(source_file["source_file"]))
```